src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp

```diff
@@ -208,31 +198,83 @@ void RequestPlaySession(const FVector* StartLocation, const FRotator* StartRotat
 
 FSlateApplication* SlateApplication = nullptr;
 
-static void RequestPlay(int mode)
+struct FPlaySettings
+{
+    EPlayModeType PlayMode;
+    int32 NumberOfClients;
+    bool bNetDedicated;
+    bool bSpawnAtPlayerStart;
+
+    static FPlaySettings UnpackFromMode(int32_t mode)
+    {
+        FPlaySettings settings = {
+            PlayModeFromInt((mode & (16 + 32 + 64)) >> 4),
+            NumberOfPlayers(mode),
+            DedicatedServer(mode),
+            SpawnAtPlayerStart(mode),
+        };
+        return settings;
+    }
+
+    static int32_t PackToMode(const FPlaySettings& settings)
+    {
+        return (settings.NumberOfClients - 1) +
+            (settings.bSpawnAtPlayerStart ? (1 << 2) : 0) +
+            (settings.bNetDedicated ? (1 << 3) : 0) +
+            (PlayModeToInt(settings.PlayMode) << 4);
+    }
+};
+
+
+static FPlaySettings RetrieveSettings(ULevelEditorPlaySettings* PlayInSettings)
+{
+    check(PlayInSettings);
+
+    FPlaySettings settings;
+    settings.PlayMode = PlayInSettings->LastExecutedPlayModeType;
+    PlayInSettings->GetPlayNumberOfClients(settings.NumberOfClients);
+#if ENGINE_MAJOR_VERSION == 4 && ENGINE_MINOR_VERSION <= 24
+    PlayInSettings->GetPlayNetDedicated(settings.bNetDedicated);
+#else
+    settings.bNetDedicated = PlayInSettings->bLaunchSeparateServer;
+#endif
+    settings.bSpawnAtPlayerStart = PlayInSettings->LastExecutedPlayModeLocation == PlayLocation_DefaultPlayerStart;
+
+    return settings;
+}
+
+static void UpdateSettings(ULevelEditorPlaySettings* PlayInSettings, const FPlaySettings& settings)
+{
+    check(PlayInSettings);
+    
+    PlayInSettings->SetPlayNumberOfClients(settings.NumberOfClients);
+#if ENGINE_MAJOR_VERSION == 4 && ENGINE_MINOR_VERSION <= 24
+    PlayInSettings->SetPlayNetDedicated(settings.bNetDedicated);
+#else
+    PlayInSettings->bLaunchSeparateServer = settings.bNetDedicated;
+#endif
+    PlayInSettings->LastExecutedPlayModeLocation =
+        settings.bSpawnAtPlayerStart
+            ? PlayLocation_DefaultPlayerStart
+            : PlayLocation_CurrentCameraLocation;
+    PlayInSettings->LastExecutedPlayModeType = settings.PlayMode;
+}
```

Comment:
Don't we need to call for `PostEditChange` and `SaveConfig` on `PlayInSettings` after changing them?